convention-plugin-test-option/src/main/kotlin/EmulatorJobsMatrix.kt

```diff
@@ -12,19 +12,69 @@ class EmulatorJobsMatrix {
             .create()
     }
 
-    fun createMatrixJsonFile(rootProject: Project) {
-        val taskList = getTaskList(rootProject = rootProject).map { it.joinToString(separator = " ") }
-        val matrix = mapOf("gradle_tasks" to taskList)
-        val jsonText = gson.toJson(matrix)
-        rootProject.layout.buildDirectory.asFile.get().also { buildDir ->
-            buildDir.mkdirs()
-            File(buildDir, "emulator_jobs_matrix.json").writeText(jsonText)
-        }
+    fun createMatrixJsonFiles(rootProject: Project) {
+        mapOf(
+            "emulator_jobs_matrix.json" to getEmulatorTaskList(rootProject = rootProject),
+            "ios_test_jobs_matrix.json" to getIosTestTaskList(rootProject = rootProject),
+            "js_test_jobs_matrix.json" to getJsTestTaskList(rootProject = rootProject),
+            "jvm_test_jobs_matrix.json" to getJvmTestTaskList(rootProject = rootProject)
+        )
+            .mapValues { entry -> entry.value.map { it.joinToString(separator = " ") } }
+            .forEach { (fileName: String, taskList: List<String>) ->
+                val matrix = mapOf("gradle_tasks" to taskList)
+                val jsonText = gson.toJson(matrix)
+                rootProject.layout.buildDirectory.asFile.get().also { buildDir ->
+                    buildDir.mkdirs()
+                    File(buildDir, fileName).writeText(jsonText)
+                }
+            }
     }
 
-    fun getTaskList(rootProject: Project): List<List<String>> =
+    fun getIosTestTaskList(rootProject: Project): List<List<String>> =
+        rootProject.subprojects.filter { subProject ->
+            listOf(
+                "firebase-analytics"
+            ).contains(subProject.name).not()
+        }.map { subProject ->
+            when (val osArch = System.getProperty("os.arch")) {
+                "x86", "i386", "ia-32", "i686",
+                "x86_64", "amd64", "x64", "x86-64" -> "${subProject.path}:iosSimulatorX64Test"
```

Comment:
Task is called `iosX64Test` for X64 and `iosX86Test` for x86 machines. No Simulator in there. Since iphones were always Arm based, there is a distinction between SimulatorArm (a simulator) and Arm (an actual iphone) that is not required for Intel architecture

Reply:
updated